lte/gateway/python/magma/subscriberdb/crypto/milenage.py

```diff
@@ -317,6 +346,76 @@ def encrypt(cls, k, buf, IV=16 * b'\x00'):
         aes_cipher = AES.new(k, AES.MODE_CBC, IV)
         return aes_cipher.encrypt(buf)
 
+    @classmethod
+    def generate_m5g_xres_star(cls, key, snni, rand, xres):
```

Comment:
add type hinting, and for all the added functions

Reply:
I've addressed the comments.

Reply:
I've addressed the comment.